test/curves.cpp

```diff
@@ -392,17 +415,21 @@ void fp3_curve_test_init(std::vector<typename Fp3CurveGroup::value_type> &points
 
     std::array<integral_type, 9> coordinates;
 
+    int p = 0;
     for (auto &point : test_set.second.get_child("point_coordinates")) {
         auto i = 0;
         for (auto &coordinate_pairs : point.second) {
             for (auto &coordinate : coordinate_pairs.second) {
                 coordinates[i++] = integral_type(coordinate.second.data());
             }
         }
-        points.emplace_back(
-            typename Fp3CurveGroup::value_type(fp3_value_type(coordinates[0], coordinates[1], coordinates[2]),
-                                               fp3_value_type(coordinates[3], coordinates[4], coordinates[5]),
-                                               fp3_value_type(coordinates[6], coordinates[7], coordinates[8])));
+        typename Fp3CurveGroup::value_type curve_element(
+                    fp3_value_type(coordinates[0], coordinates[1], coordinates[2]),
+                    fp3_value_type(coordinates[3], coordinates[4], coordinates[5]),
+                    fp3_value_type(coordinates[6], coordinates[7], coordinates[8]));
+        BOOST_CHECK_MESSAGE(curve_element.is_well_formed(), "point " << p << " is not well-formed");
```

Comment:
We can probably print the value of 'curve_element' here instead of p, doesn't it have an operator<<?

Reply:
The actual value does not matter, the position in the case is more important, which of these fails:
```
enum curve_operation_test_points : std::size_t {
    p1,
    p2,
    p1_plus_p2,
    p1_minus_p2,
    p1_mul_C1,
    p2_mul_C1_plus_p2_mul_C2,
    p1_dbl,
    p1_mixed_add_p2,
    p1_to_affine,
    p2_to_special
};
```

https://github.com/NilFoundation/crypto3-algebra/blob/29a07f2ef6228fa0c6ed59e721df2552619f1b90/test/curves.cpp#L185-L196

Reply:
ok, got it.